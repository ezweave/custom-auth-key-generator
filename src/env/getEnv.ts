import { Config, parseConfig } from '../config';

export type Env = Config;

export const getEnv = (required = true): Env => parseConfig(process.env, required);
