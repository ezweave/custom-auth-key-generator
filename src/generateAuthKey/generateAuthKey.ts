import CryotpJS from 'crypto-js';

import { CompleteArgs } from '../processArgs';

export interface CreateHash {
    (privateKey: string): (path: string) => string
}

export const createHash: CreateHash = (privateKey) => (path) => CryotpJS.enc.Base64.stringify(
    CryotpJS.HmacSHA512(path, privateKey)
);

export interface FormatHash {
    (publicKey: string, headerPrefix: string): (hash: string) => string
}

export const formatHash: FormatHash = (
    publicKey,
    headerPrefix
) => (
    hash
) => `${headerPrefix} ${publicKey}:${hash}`;

export const generateAuthKey = ({
    headerPrefix,
    path,
    privateKey,
    publicKey,
}:CompleteArgs): string => {
    const hash = createHash(privateKey)(path);
    return formatHash(publicKey, headerPrefix)(hash);
};
