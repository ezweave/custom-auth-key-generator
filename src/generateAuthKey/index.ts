import { generateAuthKey } from './generateAuthKey';

export { generateAuthKey };
