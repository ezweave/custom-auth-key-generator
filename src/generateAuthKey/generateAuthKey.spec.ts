import { getEnv } from '../env';
import { generateAuthKey } from './generateAuthKey';

describe('generateAuthKey', () => {
    it('generates a hash key', () => {
        const key = generateAuthKey({
            ...getEnv(),
            path: '/v2/applications/AP21013204737/assets/52695/filecontents',
        });
        expect(key).toMatchSnapshot();
    });
});
