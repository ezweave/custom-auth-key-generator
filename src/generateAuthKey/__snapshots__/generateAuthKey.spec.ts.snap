// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateAuthKey generates a hash key 1`] = `"Somekind of Auth PUBLIC_KEY:3kYjJ/5T6inVx+3zh9Sk/LBVSqd5XUh7vhTrLgxTNH+gwZ+TqoFopzIQsd/HUF5SiYkUhJzc/sgJ65lFeRAQ8w=="`;
